payments/voucher_manager.go

```diff
@@ -79,43 +79,45 @@ func (vm *VoucherManager) Pay(channelId types.Destination, amount *big.Int, pk [
 	return voucher, nil
 }
 
-// Receive validates the incoming voucher, and returns the total amount received so far
-func (vm *VoucherManager) Receive(voucher Voucher) (*big.Int, error) {
+// Receive validates the incoming voucher, and returns the total amount received so far as well as the amount received from the voucher
+func (vm *VoucherManager) Receive(voucher Voucher) (total *big.Int, fromVoucher *big.Int, err error) {
 	vInfo, err := vm.store.GetVoucherInfo(voucher.ChannelId)
 	if err != nil {
-		return &big.Int{}, fmt.Errorf("channel not registered: %w", err)
+		return &big.Int{}, &big.Int{}, fmt.Errorf("channel not registered: %w", err)
 	}
 
 	// We only care about vouchers when we are the recipient of the payment
 	if vInfo.ChannelPayee != vm.me {
-		return &big.Int{}, nil
+		return &big.Int{}, &big.Int{}, nil
 	}
```

Comment:
Out of scope, but this seems at odd with th ebehaviour on L86. If the recipient is incorrect, or if the channel is not registered I think we should return an error.